lib/util/git/index.ts

```diff
@@ -1122,6 +1129,54 @@ export async function fetchBranch(
   }
 }
 
+/**
+ *
+ * Configure git to be able to use the git-credential feature
+ *
+ * 0. Run the command: git config --global credential.helper 'store --file=localDir/.git-credentials'
+ * 1. Creates the file localDir/.git-credentials and fill it with `https://oauth2:<TOKEN>@<ENDPOINT>
+ *
+ */
+export async function configureCredentialHelperStore(
+  content: string
+): Promise<void> {
+  logger.debug('Configuring credential helper and git-credential file');
+  const gitCredentialsFile = upath.join(
+    GlobalConfig.get('localDir'),
+    '.git-credentials'
+  );
```

Comment:
this won't support `binarySource=docker` because this file isn't mounted to the sidecar container.

so this file should probably inside private cache folder. benefit is it's deleted after each repo.

Reply:
so in `GlobalConfig.get('cacheDir')` ?